lib/Conversion/Utils/Utils.cpp

```diff
@@ -245,12 +245,20 @@ mlir::RankedTensorType GetTypeFromTensorShape(llvm::ArrayRef<int64_t> shape,
                                      elementType, encoding);
 }
 
+static int64_t getIntegerValue(Value scalar) {
+  if (auto constOp = scalar.getDefiningOp<Torch::ConstantIntOp>()) {
+    return (int64_t)constOp.getValue();
+  }
+  return -1;
```

Comment:
You should return an `std::optional<int64_t>` instead of using a special value. Its safer to unknown values and clearer why you check for -1